mrp_lot_number_propagation/models/mrp_production.py

```diff
@@ -92,9 +91,112 @@ def _set_lot_number_propagation_data_from_bom(self):
             else:
                 propagate_move.propagate_lot_number = True
 
+    def _get_quantity_produced_issues(self):
+        if (
+            self.env.context.get("bypass_pre_button_mark_done_checks")
+            and self.is_lot_number_propagated
+        ):
+            return False
+        return super()._get_quantity_produced_issues()
+
+    def _set_quantities(self):
+        if (
+            self.env.context.get("bypass_pre_button_mark_done_checks")
+            and self.is_lot_number_propagated
+        ):
+            return True
+        return super()._set_quantities()
+
+    def _auto_production_checks(self):
+        if (
+            self.env.context.get("bypass_pre_button_mark_done_checks")
+            and self.is_lot_number_propagated
+        ):
+            return True
+        return super()._auto_production_checks()
+
     def pre_button_mark_done(self):
-        self._create_and_assign_propagated_lot_number()
-        return super().pre_button_mark_done()
+        if all(prod._auto_production_checks() for prod in self):
+            self._create_and_assign_propagated_lot_number()
+        res = super().pre_button_mark_done()
+        if isinstance(res, dict) and res["res_model"] == "mrp.batch.produce":
+            # In case a wizard action for mrp.batch.produce is returned, we have to
+            #  handle differently orders that need to go through this wizard and
+            #  those which do not
+
+            # But first let's check if actual production requiring batch producing
+            #  is set to propagate lot number
+            batch_order_id = res["context"]["default_production_id"]
+            batch_order = self.browse(batch_order_id)
+            # If it's not set to propagate lot number, return the action as it
+            #  requires user input
+            if not batch_order.is_lot_number_propagated:
+                return res
+            # Check if all batch productions are set to propagate lot number
+            productions_auto_ids = set()
+            productions_batch_ids = set()
+            for order in self:
+                if order._auto_production_checks():
+                    productions_auto_ids.add(order.id)
+                else:
+                    productions_batch_ids.add(order.id)
+
+            batch_productions = self.browse(productions_batch_ids)
+            for order in batch_productions:
+                # If any production is not set to propagate, it will require user input
+                #  so return the wizard for that order instead of one with lot number
+                #  propagation
+                if not order.is_lot_number_propagated:
+                    res["context"] = {"default_production_id": order.id}
```

Comment:
I would not override the whole ctx unless wanted
```suggestion
                    res["context"].update({"default_production_id": order.id})
```

Reply:
Actually the context return is exactly this, but I agree we can update to be nice with any other module that would override this